main.go

```diff
@@ -136,24 +137,35 @@ func (c *Client) Do(req *http.Request, v interface{}) (*http.Response, error) {
 		}
 		return nil, err
 	}
-	defer resp.Body.Close()
 
 	respErr := checkResponse(resp)
 	if respErr != nil {
 		return nil, respErr
 	}
 
-	if v != nil {
-		if w, ok := v.(io.Writer); ok {
-			io.Copy(w, resp.Body)
-		} else {
-			err := json.NewDecoder(resp.Body).Decode(v)
-			if err != nil && err != io.EOF {
-				return nil, err
-			}
-		}
+	if v == nil {
+		return resp, nil
+	}
+
+	bodyBytes, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
 	}
 
+	defer func() {
+		resp.Body.Close()
+		resp.Body = ioutil.NopCloser(bytes.NewBuffer(bodyBytes))
```

Comment:
In this way we copy 2 times the response body, but the common library use case is to access to the json response body and unmarshal it in a given structure. So this change is an unnecessary copy of a possibly large body.

What's your use case to reuse the response body?
What do you think about set this behaviour under an option flag in client creation?